src/creature.cpp

```diff
@@ -79,6 +79,17 @@ bool Creature::canSeeCreature(const Creature* creature) const
 	return true;
 }
 
+void Creature::refreshCreatureIcons() const
+{
+	SpectatorVec spectators;
+	g_game.map.getSpectators(spectators, this->getPosition(), true, true);
+	for (Creature* spectator : spectators) {
+		if (auto player = spectator->getPlayer()) {
```

Comment:
I think that here should also be verified in this way:
```cpp
assert(dynamic_cast<Player*>(spectator) != nullptr);
static_cast<Player*>(spectator)->sendUpdateCreatureIcons(this);
```

Reply:
Spectator existing and being player is already verified in `getSpectators` (`onlyPlayers` param). Would be nice if PR author could at least mention where he got that code from.

Reply:
Idk from where i took this code, it's possible it is your code :)
i have it for a while... :D

Reply:
![image](https://user-images.githubusercontent.com/17506599/229376359-e11e08ae-a5e1-4465-9bb6-b86a042d28db.png)